### PR TITLE
[WiP] magento/devdocs#: Environment variables. Add schema about overrides

### PR DESCRIPTION
**Pull request by atwixfirster**
## Purpose of this pull request

This pull request (PR) adds explanations about configuration variables override schema.

cloud-override-schema

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.2/cloud/env/variables-intro.html
- https://devdocs.magento.com/guides/v2.3/cloud/env/variables-intro.html

## Links to Magento source code

- ...

Thank you!

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/cloud/env/variables-intro.md [hunk 1]
```diff
@@ -17,4 +17,8 @@ functional_areas:
 
 Variables are _hierarchical_, which means that if a variable is not overridden, it is inherited from the parent environment.
 
+Environment variables in the top of configuration hierarchy and overrides config.php, DataBase and XML configuration variables:
```
Comment (rogyar) [inline guides/v2.2/cloud/env/variables-intro.md]:
```suggestion
Environment variables are on the top of the configuration hierarchy and override values from config.php, database, and XML configuration variables: 
```

Comment (atwixfirster) [reply guides/v2.2/cloud/env/variables-intro.md]: **fixed**

Thank you, @rogyar

**Comment by hguthrie** [comment]
@atwixfirster We store Cloud graphics in the /common/images/cloud folder.
@meker12 can you review this?

**Comment by atwixfirster** [comment]
> @atwixfirster We store Cloud graphics in the /common/images/cloud folder.

@hguthrie , thank you for notice

**fixed**

**Comment by meker12** [comment]
@atwixfirster   This information is about Magento Commerce environment variables in general, and fits better in the discussion of environment variables and scopes in the Configuration Guide:  [Use environment variables to override configuration settings](https://devdocs.magento.com/guides/v2.3/config-guide/prod/config-reference-var-name.html).  

For Cloud, we discuss store configuration settings in [Configuration management for store settings](https://devdocs.magento.com/guides/v2.3/cloud/live/sens-data-over.html).

For the Cloud Guide [environment variables](https://devdocs.magento.com/guides/v2.3/cloud/env/variables-intro.html) topic included in this PR, you can find  information about working with the environment variables for Cloud environments in the [Working with variables](https://devdocs.magento.com/guides/v2.3/cloud/env/working-with-variables.html) topic in the Cloud Guide.

I'm closing this PR because it doesn't really fit here, and the update requires more investigation to determine the best place for the information.  Before submitting a PR to add significant new content that's more complex, start by creating a [GitHub issue](https://github.com/magento/devdocs/issues/new/choose) with your proposal before making the support updates and submtting the PR. See the [DevDocs contribution guidelines](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md#contribution-guidelines).

**Comment by atwixfirster** [comment]
Thank you for the review, @meker12 !

However , just to finalize here: if you want to contribute to Cloud then you should create an issue and provides your propositions there?

Thanks

**Comment by meker12** [comment]
> However , just to finalize here: if you want to contribute to Cloud then you should create an issue and provides your propositions there?

Right, if you are proposing new content or a new section.  If you are making small changes or simple updates, submitting the PR directly works.

**Comment by atwixfirster** [comment]
> Right, if you are proposing new content or a new section. If you are making small corrections to text or examples, or simple updates, submitting the PR directly works.

Thank you for the explanations, Margaret!

I do not push on you but I want to clarify here. Current PR contains a small sentence and 1 image. 

Per my understanding it's a simple update. And I do not see the reason to create a new issue for such modifications. 

So, how a contributor can understand that for Cloud changes he should create an issue (in case of a new content or a new section). This information is not present in the [DevDocs contribution guidelines](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md#contribution-guidelines).

Thank you!

**Comment by meker12** [comment]
In this case, the content you submitted isn't in the correct place, which is why I closed it.

**Comment by jeff-matthews** [comment]
I'm going to reopen this because I think it might be useful moving it to the Configuration Guide as @meker12 suggested and I want to provide @atwixfirster a chance to do that.

The style of the graphic/diagram that is being submitted will need some work before we can accept it, but we can have a separate conversation about that. We need to ensure that it meets our internal style standards.

For now, I think it's more important to see if it fits in https://devdocs.magento.com/guides/v2.3/config-guide/prod/config-reference-var-name.html.

**Comment by atwixfirster** [comment]
> I'm going to reopen this because I think it might be useful moving it to the Configuration Guide as @meker12 suggested and I want to provide @atwixfirster a chance to do that.

Thank you, @jeff-matthews 

**The main idea of this PR:** show for readers the readable (easy understandable) information about configuration variables overrides.

@meker12 @jeff-matthews  I am contributor. Now, I do not fight for the contributors points. Now, I am fight for the contribution as a delivery process.

I've created this PR to add something important (from my perspective) to the existing documentation. If information provided in the PR is not unnecessary information then let's close this PR. But if you really think that a mentioned diagram explains override variable principles better than pile of unreadable text, then let's together find the place where it can be utilized, so it can be useful and used by readers.

> The style of the graphic/diagram that is being submitted will need some work before we can accept it, but we can have a separate conversation about that. We need to ensure that it meets our internal style standards.

I agreed that image is not a super cool but I can draw it the color schema which you will requested.

Thank you!

**Comment by meker12** [comment]
**Do not Merge.** This PR is on hold for further investigation about how to fit it into the Configuration Guide topic referenced in the PR comments.

**Comment by meker12** [comment]
@atwixfirster  Do you agree that we should close this branch, and then start a new one to add the changes?  Perhaps, opening an issue about the topic where we can discuss the updates, and then creating a PR based on that issue.

**Comment by atwixfirster** [comment]
>  Do you agree that we should close this branch

I agreed, @meker12 

Thanks

**Comment by dobooth** [comment]
Closed as above.

**Pull request closed without merging**